File: src/app/src/infrastructure/redis.js
const IORedis = require("ioredis");

class Redis {
    constructor() {
        this.redis = new IORedis({
            host: process.env.REDIS_HOST || "localhost",
            port: process.env.REDIS_PORT || 6379,
        });
    }


    async get(key) {
        const value = await this.redis.get(key);
        return value ? JSON.parse(value) : null;
    }

    set(key, value, timeExp) {
        return this.redis.set(key, JSON.stringify(value), "EX", timeExp);
    }
}
module.exports = new Redis();

File: db/redis.py
import redis
import os



def connect_to_redis():
    host = os.environ.get('DB_HOST', 'localhost')
    port = os.environ.get('DB_PORT', 6379)
    db = os.environ.get('DB_NUMBER', 0)
    return redis.Redis(host=host, port=port, db=db)
